### PR TITLE
Add support for typedef and unions

### PR DESCRIPTION
**Pull request by TimothyGu**

### CODE REVIEW

**Comment by domenic** [comment]
Wow. This is extraordinary work, based on just skimming the volume of code and the very-correct-seeming output in the snapshot file. I hope to have time to review in more detail soon; is there anything in particular you'd like feedback on? It's clear at this point you know this codebase better than most :)

**Comment by TimothyGu** [comment]
> is there anything in particular you'd like feedback on?

I don't think so, not on this PR at least. 

> It's clear at this point you know this codebase better than most :)

Well, just the JS-to-IDL type conversion part really :D

**Comment by TimothyGu** [comment]
@domenic ping. Would you mind reviewing/applying this?

**Comment by stevenvachon** [comment]
@domenic still sick?

File: lib/output/utils.js [hunk 1]
```diff
@@ -48,12 +68,18 @@ function tryImplForWrapper(wrapper) {
...
+  wrapperSymbol,
+  implSymbol,
+  getSameObject,
+  wrapperForImpl,
+  implForWrapper,
+  tryWrapperForImpl,
+  tryImplForWrapper,
+  iterInternalSymbol,
+  IteratorPrototype
+};
```
Comment (ExE-Boss) [inline lib/output/utils.js]: Please don't reformat existing files for no particular gains.

Comment (domenic) [reply lib/output/utils.js]: @ExE-Boss you are not a reviewer for this project, so please do not impose your style advice (which often conflicts with the maintainers) on others. We've been mostly able to ignore your reviews so far but if you are rude, then we'll have to take larger measures.

**Pull request merged**